### PR TITLE
Test parlai/core/script.py

### PR DESCRIPTION
**Pull request by stephenroller**
**Patch description**
Add tests for the abstract class in script.py, and the supercommand.

**Testing steps**
New CI.
```
$ coverage report parlai/core/script.py
Name                    Stmts   Miss  Cover
-------------------------------------------
parlai/core/script.py     137      2    99%
```

### CODE REVIEW

**Review by klshuster — APPROVED** [review]
big fan of this, hopefully this will catch the super subtle differences that sometimes come up between e.g. `parlai tm` and `python -m parlai.scripts.train_model` - ideally we can continue adding tests to this for maximum robustness

File: tests/test_script.py [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+    def test_supercommand(self):
+        opt = script.superscript_main(args=['test_script', '--foo', 'test'])
+        assert opt.get('foo') == 'test'
+
+    def test_no_setup_args(self):
+        with self.assertRaises(NotImplementedError):
+            script.superscript_main(args=['no_setup_args'])
+
+    def test_help(self):
+        with testing_utils.capture_output() as output:
```
Comment (jxmsML) [inline tests/test_script.py:91]: is this redundant w.r.t. L101 - 104

**Review by jxmsML — APPROVED** [review]

File: tests/test_script.py (hunk 1 above)
Comment (stephenroller) [reply tests/test_script.py:91]: lol yes it is, will drop

File: tests/test_script.py [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+        assert opt.get('foo') == 'test'
+        assert opt.get('bar') == 'sneaky'
+
+    def test_main_args(self):
+        opt = _TestScript.main('--foo', 'test')
+        assert opt.get('foo') == 'test'
+        assert opt.get('bar') == 'sneaky'
+
+    def test_main_noargs(self):
+        with patch.object(sys, 'argv', ['test_script.py']):
```
Comment (stephenroller) [inline tests/test_script.py:57]: wait why does this work

Comment (stephenroller) [reply tests/test_script.py:57]: ah, argv[0] doesn't matter
